src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -156,54 +172,59 @@ public Command intake() {
 			elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).handleInterrupt(() -> enableChangeStateAutomatically(true).schedule());
 	}
 
 	public Command preSpeaker() {
 		return new ParallelCommandGroup(
 			setCurrentStateValue(RobotState.PRE_SPEAKER),
+			enableChangeStateAutomatically(true),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER)),
 			elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
 			flywheelStateHandler.setState(FlywheelState.SHOOTING),
 			funnelStateHandler.setState(FunnelState.STOP),
 			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).handleInterrupt(() -> enableChangeStateAutomatically(true).schedule());
 	}
 
 	public Command speaker() {
 		return new ParallelCommandGroup(
 			setCurrentStateValue(RobotState.SPEAKER),
+			enableChangeStateAutomatically(false),
 			new SequentialCommandGroup(
 				funnelStateHandler.setState(FunnelState.STOP).until(this::isReadyToShoot),
 				funnelStateHandler.setState(FunnelState.SPEAKER).until(() -> !isNoteInShooter()),
 				funnelStateHandler.setState(FunnelState.STOP)
 			),
+			enableChangeStateAutomatically(true),
```

Comment:
I don't think this should be here, think about this:
The parallel command starts, and the sequential runs at the same time as this, not before, so right at the beginning of the command, it will set to true and u move to idle so nothing will happen